### PR TITLE
Add endpoints for create tf and user

### PR DESCRIPTION
**Pull request by romeufcrosa**

### CODE REVIEW

File: cmd/api/api.go [hunk 1]
```diff
@@ -5,19 +5,27 @@ import (
...
 
+var (
+	infra bootstrap.Infrastructure
+	err   error
+)
+
 func main() {
 	port := os.Getenv("PORT")
-	infra, err := bootstrap.NewInfrastructure("stg")
+	infra, err = bootstrap.NewInfrastructure("stg")
```
Comment (miguel-botelho) [inline cmd/api/api.go:24]: Why do we need the `infra` variable?

File: cmd/api/api.go [hunk 2]
```diff
@@ -35,41 +43,119 @@ func apiKeyMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
...
+	taskforceID, err := uuid.Parse(c.Param("taskforce_id"))
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, requests.SchedulerBadResponse{Code: "0610001", Message: err.Error()})
+	}
+
+	if err := infra.Services.Taskforce.AddUser(*request, taskforceID); err != nil {
+		return c.JSON(http.StatusInternalServerError, requests.SchedulerBadResponse{Code: "0610002", Message: err.Error()})
+	}
+
+	return c.String(http.StatusCreated, "User added successfully to task")
```
Comment (miguel-botelho) [inline cmd/api/api.go:103]: Shouldn't this return a JSON payload?

File: cmd/api/api.go [hunk 3]
```diff
@@ -35,41 +43,119 @@ func apiKeyMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
...
+	if err := c.Bind(request); err != nil {
+		return c.JSON(http.StatusInternalServerError, requests.SchedulerBadResponse{Code: "0610001", Message: err.Error()})
+	}
+
+	taskforce, err := infra.Services.Taskforce.Create(*request)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, requests.SchedulerBadResponse{Code: "0610002", Message: err.Error()})
+	}
+
+	return c.String(http.StatusCreated, fmt.Sprintf("Created task %s", taskforce.ID.String()))
```
Comment (miguel-botelho) [inline cmd/api/api.go:117]: And this one as well

File: cmd/api/api.go (hunk 1 above)
Comment (romeufcrosa) [reply cmd/api/api.go:24]: It's used in the handler functions

File: cmd/api/api.go (hunk 2 above)
Comment (romeufcrosa) [reply cmd/api/api.go:103]: Return the object created in JSON.
Yes it makes more sense, I was lazy. I'll change it.

File: cmd/api/api.go (hunk 3 above)
Comment (romeufcrosa) [reply cmd/api/api.go:117]: Will change it 👍

File: cmd/api/api.go (hunk 1 above)
Comment (romeufcrosa) [reply cmd/api/api.go:24]: I move it to the package level, to be able to extract the logic of the endpoint definition to separate functions
This way I avoid having to pass `infra` to all of them.

The value is assigned here in the constructor and then used everywhere.
This is analog to Ruby's `@` variables

**Review by miguel-botelho — APPROVED** [review]

**Pull request merged**